src/transformers/models/maskformer/configuration_maskformer.py

```diff
@@ -0,0 +1,179 @@
+# coding=utf-8
+# Copyright 2022 Facebook AI Research and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" MaskFormer model configuration"""
+import copy
+from typing import Dict, Optional
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from ..auto.configuration_auto import AutoConfig
+from ..detr import DetrConfig
+from ..swin import SwinConfig
+
+
+MASKFORMER_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "Francesco/maskformer-swin-base-ade": "https://huggingface.co/Francesco/maskformer-swin-base-ade/blob/main/config.json"
+    # See all MaskFormer models at https://huggingface.co/models?filter=maskformer
+}
+
+logger = logging.get_logger(__name__)
+
+
+class MaskFormerConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MaskFormer`]. It is used to instantiate a
+    MaskFormer model according to the specified arguments, defining the model architecture. Instantiating a
+    configuration with the defaults will yield a similar configuration to that of the
+    "Francesco/maskformer-swin-base-ade" architecture trained on ADE20k-150
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Currently, maskformer supports only Swin backbone.
+
+    Args:
+        mask_feature_size (`int`, *optional*, defaults to 256):
+            The masks' features size, this value will also be used to specify the Feature Pyramid Network featuresc
+            size.
+        no_object_weight (`float`, *optional*, defaults to 0.1):
+            Weight to apply to the null class .
+        use_auxilary_loss (`bool`, *optional*, defaults to `False`):
+            If `true` [`MaskFormerOutput`] will contain.
+        backbone_config (`Dict`, *optional*):
+            The configuration passed to the backbone, if unset, the configuration corresponding to `swin-base` will be
+            used.
+        detr_config (`Dict`, *optional*):
+            The configuration passed to the Detr model, if unset the base config for `detr` will be used.
```

Comment:
```suggestion
            The configuration passed to the Detr model, if unset the base config for `detr` will be used.
```
'detr' is also not an architecture on the hub, do you mean `facebook/detr-resnet-50`?

Reply:
Resolved, keep up the other conversation bout DETR vs Transformer Decoder